stumpy/core.py

```diff
@@ -589,6 +593,37 @@ def check_window_size(m, max_size=None):
     if max_size is not None and m > max_size:
         raise ValueError(f"The window size must be less than or equal to {max_size}")
 
+    if n is not None:
```

Comment:
@seanlaw 
Can you please read the comments and let me know if the logic is clear?

Reply:
Please ignore for now as I've changed the approach.